### PR TITLE
post categories

### PR DESCRIPTION
**Pull request by MostafaQanoo**

### CODE REVIEW

**Review by Mu7ammadAbed — COMMENTED** [review]
Add more tests for the edge cases.

File: test/index.test.ts [hunk 1]
```diff
@@ -212,6 +212,20 @@ describe('Test post categories', () => {
...
+    const res = await supertest(app)
+      .post('/api/v1/categories')
+      .send({
+        name: 'mostafa',
+        description: '',
+        image: 'dsfdsfsdggs',
+      })
+      .expect(400)
+      .expect('Content-Type', /json/);
+    expect(res.body.status).toBe(400);
```
Comment (Mu7ammadAbed) [inline test/index.test.ts:226]: You already tested the status code in line `224`, try to test the response message

**Review by Mu7ammadAbed — CHANGES_REQUESTED** [review]

**Review by Mu7ammadAbed — APPROVED** [review]

**Review by Mu7ammadAbed — APPROVED** [review]

**Review by Jehad91 — APPROVED** [review]

**Review by MohammadAlHabil — APPROVED** [review]

**Review by IsraaHamdi — APPROVED** [review]
